A28-xds-traffic-splitting-and-routing.md

```diff
@@ -0,0 +1,1095 @@
+    *   Because this will only have CDS as children, this is essentially
+        `weighted_CDS`
+
+More child policies can be added in the future.
+
+### Balancer config
+
+The `xds_routing_experimental` LB policy will have the following
+configuration:
+
+```proto
+/*
+
+Route List                         Action list
++---------+--------+               +-------+-----------------+--------------+
+| matcher | action |               | name  | policy          | config       |
++------------------+               +----------------------------------------+
+| /s/m1   | wt_1   | +-----------> | wt_1  | weighted_target | <config_1>   |
++------------------+               +----------------------------------------+
+| /s/m2   | cds_1  | +----\ /----> | wt_2  | weighted_target | <config_2>   |
++------------------+       x       +----------------------------------------+
+| /s2/m3  | wt_2   | +----/ \----> | cds_1 | CDS             | <cds_config> |
++------------------+            -> +-------+-----------------+--------------+
+| /s2/m4  | cds_1  | +---------/
++---------+--------+
+
+ */
+
+message XdsRouting {
+  message Action {
+    repeated LoadBalancingConfig child_policy = 1;
+  }
+  // Route represents a routing rule to send an RPC. It contains a matcher and
+  // the name of the action to take. The action with the name can be found in
+  // the action list.
+  //
+  // Route contains the action name instead of embedding the Action, so Action
+  // can be shared and reused by multiple routes.
+  message Route {
+    message HeaderMatcher {
+      message Int64Range {
+        int64 start = 1;
+        int64 end = 2;
+      }
+      string name = 1;
+      oneof header_match_specifier {
+        string exact_match = 2;
+        string regex_match = 3;
+        Int64Range range_match = 4;
+        bool present_match = 5;
+        string prefix_match = 6;
+        string suffix_match = 7;
+      }
+      bool invert_match = 8;
+    }
+    message Fraction {
+      // Specifies the numerator. Defaults to 0.
+      uint32 numerator = 1;
+
+      // Specifies the denominator. If the denominator specified is
+      // less than the numerator, the final fractional percentage
+      // is capped at 1 (100%).
+      uint32 denominator = 2;
+    }
+
+    oneof path_matcher {
+      string path = 1;
+      string prefix = 2;
+      string regex = 3;
+    }
+    repeated HeaderMatcher headers = 3;
+
+    // Every time the route is considered for a match, it
+    // must also fall under the percentage of matches
+    // indicated by this field. For some fraction N/D, a
+    // random number in the range [0,D) is selected. If the
+    // number is <= the value of the numerator N, or if the
+    // key is not present, the default value, the router
+    // continues to evaluate the remaining match criteria.
+    Fraction match_fraction = 4;
+
+    string action = 5; // The action name
+  }
+
+  repeated Route route = 1;
+  map<string, Action> action = 2;
```

Comment:
A comment to explain that this map maps an action name to the actual action will be helpful.

Reply:
Done